### DISCUSSION TITLE
Is this install OK? (I get some errors)

### DISCUSSION THREAD

**Discussion by navatweb** [open-policy-agent/community#549] [Gatekeeper]
kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml
===

**First try:**
[Yesterday 2:24 PM] Navaneet Kumar19
# kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml

namespace/gatekeeper-system created

resourcequota/gatekeeper-critical-pods created

customresourcedefinition.apiextensions.k8s.io/assign.mutations.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/assignimage.mutations.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/assignmetadata.mutations.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/configs.config.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/constraintpodstatuses.status.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/constrainttemplatepodstatuses.status.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/constrainttemplates.templates.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/expansiontemplate.expansion.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/expansiontemplatepodstatuses.status.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/modifyset.mutations.gatekeeper.sh created

customresourcedefinition.apiextensions.k8s.io/mutatorpodstatuses.status.gatekeeper.sh created

error: error validating "[https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml":](https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml%22:) error validating data: [ValidationError(CustomResourceDefinition.spec.versions[0]): unknown field "deprecated" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion, ValidationError(CustomResourceDefinition.spec.versions[0]): unknown field "deprecationWarning" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion]; if you choose to ignore these errors, turn validation off with --validate=false

**Second try:**

# kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml --validate=false
[Yesterday 2:38 PM] Navaneet Kumar19
root@lxco:~# kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml --validate=false

namespace/gatekeeper-system unchanged

resourcequota/gatekeeper-critical-pods unchanged

customresourcedefinition.apiextensions.k8s.io/assign.mutations.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/assignimage.mutations.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/assignmetadata.mutations.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/configs.config.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/constraintpodstatuses.status.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/constrainttemplatepodstatuses.status.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/constrainttemplates.templates.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/expansiontemplate.expansion.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/expansiontemplatepodstatuses.status.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/modifyset.mutations.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/mutatorpodstatuses.status.gatekeeper.sh configured

customresourcedefinition.apiextensions.k8s.io/providers.externaldata.gatekeeper.sh created

serviceaccount/gatekeeper-admin created

role.rbac.authorization.k8s.io/gatekeeper-manager-role created

clusterrole.rbac.authorization.k8s.io/gatekeeper-manager-role created

rolebinding.rbac.authorization.k8s.io/gatekeeper-manager-rolebinding created

clusterrolebinding.rbac.authorization.k8s.io/gatekeeper-manager-rolebinding created

secret/gatekeeper-webhook-server-cert created

service/gatekeeper-webhook-service created

deployment.apps/gatekeeper-audit created

deployment.apps/gatekeeper-controller-manager created

mutatingwebhookconfiguration.admissionregistration.k8s.io/gatekeeper-mutating-webhook-configuration created

validatingwebhookconfiguration.admissionregistration.k8s.io/gatekeeper-validating-webhook-configuration created

error: unable to recognize "[https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml":](https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml%22:) no matches for kind "PodDisruptionBudget" in version "policy/v1"

**Comment by navatweb** [comment]
In my Cluster I don't have PodDisruptionBudget. I downloaded the yaml  "[https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml"](https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v3.14.0/deploy/gatekeeper.yaml%22:)
and commented out that the section on PodDisruptionBudget. Did kubectl -f apply <yaml> and the installation completed without error.
I hope I don't have to use PodDisruptionBudget

**Comment by ritazh** [comment]
Please take a look at installing with helm as it has many more conditional options for you to choose from as every environment is quite different. e.g https://github.com/open-policy-agent/gatekeeper/blob/master/charts/gatekeeper/templates/gatekeeper-controller-manager-poddisruptionbudget.yaml gets installed only if the cluster has the pdb capability.
